crates/sqllib/src/casts.rs

```diff
@@ -132,126 +232,130 @@ cast_to_b!(u, usize);
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_b_s(value: String) -> bool {
-    value.trim().parse().unwrap_or(false)
+pub fn cast_to_b_s(value: String) -> SqlResult<bool> {
+    Ok(value.trim().parse().unwrap_or(false))
```

Comment:
I'd not expect the cast to just change my result to `false` if it fails but maybe it can never happen?

Reply:
this deserves more tests, I am under the impression that this is what Calcite does. We have to match its behavior when it constant folds.